src/openrct2/ride/Ride.cpp

```diff
@@ -2595,6 +2595,10 @@ bool Ride::CanBreakDown() const
 
 static void choose_random_train_to_breakdown_safe(Ride* ride)
 {
+    // Prevent integer division by zero in case of hacked ride.
+    if (ride->num_vehicles == 0)
```

Comment:
What happens when this function returns without setting a broken_vehicle. The callee does not know the function has failed. This could cause unexpected results.